### PR TITLE
fix: eth to weth on tokenlist

### LINKED ISSUE #1458: For Swapr on Polygon - Asset should be "WETH" not "ETH"
**Description**
For Swapr on Polygon - Asset should be "WETH" not "ETH"

Unable to find "WETH" when searching assets

image

Here is the asset:
https://polygonscan.com/token/0x7ceb23fd6bc0add59e62ac25578270cff1b9f619?a=0xe04885c3f1419c6e8495c33bdcf5f8387cd88846

### PR DESCRIPTION
**Pull request by berteotti**
# Summary

Fixes #1458

Rename ETH to WETH.

image

### CODE REVIEW

File: src/hooks/Tokens.ts [hunk 1]
```diff
@@ -31,7 +31,16 @@ function useTokensFromMap(tokenMap: TokenAddressMap, includeUserAdded: boolean):
 
     // reduce to just tokens
     const mapWithoutUrls = Object.keys(tokenMap[chainId]).reduce<{ [address: string]: Token }>((newMap, address) => {
-      newMap[address] = tokenMap[chainId][address].token
+      let token = tokenMap[chainId][address].token
+
+      if (token.equals(WETH[chainId]))
```
Comment (Diogomartf) [inline src/hooks/Tokens.ts:36]: why does tokenMap does not have the correct form of weth?

Comment (berteotti) [reply src/hooks/Tokens.ts:36]: Because the list is wrong in this case. Ideally this should be changed on the token list but we don't control that.

**Comment by niemam29** [comment]
Tested and looks good to me

File: src/hooks/Tokens.ts (hunk 1 above)
Comment (adamazad) [inline src/hooks/Tokens.ts:36]: I'm always hesitant about this instance comparison. I prefer checksummed address comparison. This is still sufficient for our use case now :)

**Review by adamazad — APPROVED** [review]
Works as described. Nice work!

**Pull request merged**